### ISSUE TITLE
The Ruff VS Code extension uses the wrong `ruff.toml` file

### ISSUE THREAD

**Issue by ebbek** [astral-sh/ruff#28483]
### Summary

I have a bunch of projects that do not follow the PEP8 formatting rules (longer lines and non-standard indentation). For these I have made custom `ruff.toml` file that is referred to in the respective `pyproject.toml` files like this:
```
[tool.ruff]
extend = "~/projects/notpep8/ruff.toml"
```

The file structure is like this:
```
~/projects --- notpep8 -+- ruff.toml
                        |
                        +- proj -+- pyproject.toml
                        |        |
                        |        +- src --- example.py
```

This works perfectly when running `ruff check` from the command line in the respective project directories. 

However when running ruff in VS Code using that Astral ruff extension, this seems to be ignored and instead the global PEP8 compliant `ruff.toml` file is used. This of course produces loads of errors.

I have tried several different things:

- Made an intentional error in pyproject.toml. This is detected so ruff does look at the file.
- Added `"ruff.configuration": "~/projects/notpep8/ruff.toml"` lines to the various `settings.json` files. That does not change anything.

As far as I can see from the attached log file ruff does seem to load the local `ruff.toml` file - and then it loads the global one immediately afterwards.

[ruff.log](https://github.com/user-attachments/files/32057870/ruff.log)

### Version

ruff 0.16.6 (22f65a2ab 2026-09-03)

**Comment by MichaReiser** [comment]
Hi. 

I tried to reproduce this on linux and failed. This could be a windows specific issue, but reproducing it would be easier if you could create a small reproduction. Ideally a single zip with:

* Your shared configuration (can be a downsized version of it that only configures the PEP8 relevant rules)
* An `example.py` that contains code that Ruff incorrectly flags (in VS Code but not when using `ruff check`)
* The Project's `pyproject.toml`

Thank you

**Comment by ebbek** [comment]
I have created the attached quick-and-dirty example in the `notpep8.zip` file. It must be unpacked in the directory `~/projects`.

The `ruff.zip` file contains a copy of my global `ruff.toml` file

VS Code shall have at least the Python and Ruff extensions installed and it shall be started from the `project`directory like this:

```
code .\.vscode\project.code-workspace &
```

[notpep8.zip](https://github.com/user-attachments/files/32062203/notpep8.zip)

[ruff.zip](https://github.com/user-attachments/files/32062307/ruff.zip)

**Comment by MichaReiser** [comment]
Thank you. Unfortunately, I'm still unable to reproduce. I'll try go get hold of a Windows machine to see if I can reproduce it there. But in the meantime:

Could you open `src/example.py` in the reproduction workspace, run `Ruff: Print debug information` (native server only) from the Command Palette, and share the output? Please review the output for private paths before posting.

**Comment by ebbek** [comment]
Done:

[ruffdebug.txt](https://github.com/user-attachments/files/32086145/ruffdebug.txt)

And by the way: I have tried it on Linux  now and I too cannot reproduce the problem on that platform.

Edit: It also works as expected when using the windows version of ´code´ in WSL.

**Comment by MichaReiser** [comment]
Thank you. I'll take another look. 

> And by the way: I have tried it on Linux now and I too cannot reproduce the problem on that platform.

This is good to know. I'll try to reproduce on a windows machine

**Comment by MichaReiser** [comment]
The logs suggest that there's a `ruff.toml` in the `project` folder

```
ndexed_configuration_files = [
    "c:\\Users\\user\\projects\\notpep8\\ruff.toml",
    "c:\\Users\\user\\projects\\notpep8\\project\\ruff.toml",
]
```

What does it contain? I'm asking because a `ruff.toml` takes precedence over a `pyproject.toml`

**Comment by ebbek** [comment]
Indeed - and there shouldn't be a `ruff.toml` there. However, after I have removed it, the problem persists. I have attached the updated version of ``ruffdebug.txt.

[ruffdebug.txt](https://github.com/user-attachments/files/32089495/ruffdebug.txt)

I should mention that I will be away from my PC from now on and until Monday morning CET so right now there is no haste from my side in getting this solved.

**Comment by MichaReiser** [comment]
Hmm, that's strange. The logs now show that Ruff uses the longer line length

```
linter.tab_size = 3
linter.line_length = 140
linter.task_tags = [
	TODO,
	FIXME,
	XXX,
]
```

Which I believe is what you expect based on your shared ruff config. 

Can you share the output of running `ruff check`, maybe by running `ruff check -v` (including in which directory you run the command)

**Comment by ebbek** [comment]
Indeed - and there shouldn't be a `ruff.toml` there. However, after I have removed it, the problem persists. I have attached the updated version of `ruffdebug.txt` and the output I get when running from the command line in `ruffcmdline.txt`.

[ruffdebug.txt](https://github.com/user-attachments/files/32089495/ruffdebug.txt)

[ruffcmdline.txt](https://github.com/user-attachments/files/32181927/ruffcmdline.txt)

**Comment by ebbek** [comment]
I must apologize for you wasting your time on this.

This is a case of me not looking at what is actually shown on the screen. The "problem persists" was caused by the pylint vscode extension and the error messages disappeared once I disabled that extension.

So there is no problem with the ruff vscode extension and you may close this issue. If you have an "irrelevant" label or something like it, I suggest you apply it to this issue 😳

**Comment by MichaReiser** [comment]
No worries and I'm glad to hear that it's now working for you. I should have asked earlier if you have any other extensions installed. 

Thanks for posting an update. I'll close this issue as completed.
